src/isl/mod.rs

```diff
@@ -310,10 +317,10 @@ mod isl_tests {
                     "#),
         IslType::anonymous(
             [IslConstraint::valid_values_with_range(
-                Range::number_range(
-                    RangeBoundaryValue::Value((&IntegerValue::I64(1)).into(), RangeBoundaryType::Inclusive),
-                    RangeBoundaryValue::Value((&Decimal::new(55, -1)).try_into().unwrap(), RangeBoundaryType::Inclusive)
-                ).unwrap())
+                NumberRange::new(
+                    Number::from(&IntegerValue::I64(1)).into(),
+                    Number::try_from(&Decimal::new(55, -1)).unwrap().into()
```

Comment:
I was curious why `&Integer` could be converted to a `Number` infallibly while `&Decimal` had to use `try_from`. Looking at the source:

https://github.com/amzn/ion-schema-rust/blob/b5b643593dc1260e4048531f1ef61ff391d31066/src/isl/util.rs#L596-L604

`Decimal` cannot be converted to a `BigDecimal` if it's a negative zero, but will succeed for all other values. Since `Number` is being used in ranges and for ordering, I think its implementation could detect negative zero and return `BigDecimal::zero()` instead, making it infallible. What do you think?

Reply:
Yes that makes sense